components/brave_ads/core/internal/ads/serving/choose/sample_ads.h

```diff
@@ -16,16 +17,28 @@
 
 namespace ads {
 
+template <typename T>
+T CalculateNormalizingConstantBase(const std::vector<T> scores) {
+  T normalizing_constant = 0.0;
+  for (const auto& votes : scores) {
+    normalizing_constant += votes;
+  }
+
+  return normalizing_constant;
+}
+
 template <typename T>
 double CalculateNormalizingConstant(
     const CreativeAdPredictorMap<T>& creative_ad_predictors) {
-  double normalizing_constant = 0.0;
+  std::vector<double> scores;
+  scores.assign(creative_ad_predictors.size(), 0);
 
-  for (const auto& [segment, ad_predictor] : creative_ad_predictors) {
-    normalizing_constant += ad_predictor.score;
+  for (const auto& creative_ad_predictor : creative_ad_predictors) {
```

Comment:
Can we please use structured bindings, `[creative_instance_id, segment]` I believe in this case, thanks